src/stdio.jl

```diff
@@ -24,6 +24,9 @@ if VERSION >= v"0.7.0-DEV.1472" # Julia PR #23271
     Base.unwrapcontext(io::IJuliaStdio) = Base.unwrapcontext(io.io)
 end
 Base.setup_stdio(io::IJuliaStdio, readable::Bool) = Base.setup_stdio(io.io.io, readable)
+Base.redirect_stdout(io::IJuliaStdio) = (eval(Base, :(STDOUT = $io)); io)
+Base.redirect_stderr(io::IJuliaStdio) = (eval(Base, :(STDERR = $io)); io)
+Base.redirect_stdin(io::IJuliaStdio) = (eval(Base, :(STDIN = $io)); io)
```

Comment:
The documentation seems to be wrong in this case. Calling `redirect_stdout(STDERR)`, for example, just returns `STDERR`. See: https://github.com/JuliaLang/julia/blob/v0.6.2/base/stream.jl#L1031

Reply:
https://github.com/JuliaLang/julia/issues/26558

Reply:
I guess it's only called for restoring the original STDOUT stream in `redirect_*`.   In that case, however, probably check that this is the case.  Something like:
```jl
for s in ("stdout", "stderr", "stdin")
    f = Symbol("redirect_", s)
    ex = Expr(:=, Symbol(uppercase(s)), :io)
    @eval function Base.$f(io::IJuliaStdio)
        io[:jupyter_stream] != $s && error("expecting ", $s, " stream)
        eval(Base, $ex)
        return io, io
    end
end
```

Reply:
Sure, we can do that. It does seem a bit odd for the return type to be inconsistent with the Base implementation of `redirect_stdout(STDOUT)`, though, doesn't it?

Reply:
Oh, I didn't notice that `redirect_stdout(STDOUT)` returned only a single stream.  That seems like a bug in base, since `redirect_stdout` is documented to return two streams.

But sure, in that case do `return io` rather than `return io, io`.  But we should still check that it is restoring the correct stream.

Reply:
Ok, done